collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java

```diff
@@ -128,6 +134,44 @@ public boolean canAddRepeat() {
         }
     }
 
+    public void moveForward() {
+        try {
+            formController.stepToNextScreenEvent();
+        } catch (JavaRosaException e) {
+            error.setValue(e.getCause().getMessage());
+            return;
+        }
+
+        formController.getAuditEventLogger().flush(); // Close events waiting for an end time
+    }
+
+    public void moveBackward() {
+        if (formController.getEvent() != FormEntryController.EVENT_BEGINNING_OF_FORM) {
```

Comment:
Why doesn't this also move to `moveScreen`?

Reply:
Ah yeah I got distracted by the search/rotate problem and didn't clean this up. Will move it around today.

Reply:
Realized we don't actually need this check at all so avoiding have it anywhere.